Mathlib/Tactic/Replace.lean

```diff
@@ -13,7 +13,7 @@ syntax "replace " haveDecl : tactic
 
 /--
 Acts like `have`, but removes a hypothesis with the same name as
-this one. For example, if the state is:
+this one if possible. For example, if the state is:
```

Comment:
```suggestion
this one if present. For example, if the state is:
```
I'm out of my comfort zone here, so my suggestion could be garbage.  Do you mean that if there is a hypothesis with the given name it clears it?

Reply:
Nope. Sometimes it's not possible to clear the old hypothesis:
```lean
example {a : Nat} : a = a := by
  replace a : Int := 0
  have : Nat := by assumption -- old `a` is not gone
  have : Int := by exact a    -- new `a` is of type `Int`
  simp
```
`a` is used in the goal and thus cannot be cleared out. In Lean 4 it will survive as `a†`.

Reply:
Ah, ok!  I was only thinking of what might happen if you tried to replace using a name that is not actually present in context.